### DISCUSSION TITLE
tRPC support

### DISCUSSION THREAD

**Discussion by CoenWarmer** [directus/directus#13608] [Feature Requests]
Just came across tRPC, an alternative to REST and GraphQL APIs which aims to offer a way to communicate between client and server in a type safe way without client generation. 

Links: 
- https://trpc.io
- https://blog.logrocket.com/build-full-stack-typescript-app-trpc-react/

I haven't used it before, but the idea of consuming Directus' APIs without OAS spec client generation does sound very appealing. Perhaps this is interesting to track as either a plug-in or a core feature.

**Comment by drnachio** [comment]
We have been using tRPC for a long time and today it has become one of the pieces that most accelerates the daily development of my team.

Not only does it allow the generation of a type contract between the client and the server, guaranteeing that API modifications do not affect its clients, but thanks to its ability to group requests and additional functionalities, it helps us improve the performance of our SSR with Next.


To use it with Directus we sometimes have to generate an intermediate tRCP "proxy" with the Directus JS-SKD (having to manually maintain the types of each of the methods).


Although on occasions when we want to focus on consulting without using Directus permissions, or the auditing system, or flows, etc... we opt for a tRCP API server with prisma.io as ORM pointing to the same databases what Directus.

As you can imagine avoiding these steps would be a huge help.

**Reply by rijkvanzanten** [reply]
> To use it with Directus we sometimes have to generate an intermediate tRCP "proxy" with the Directus JS-SKD (having to manually maintain the types of each of the methods).

Do you have an example of what this proxy looks like? It might give us a head start on how something like this can be implemented in the core codebase

**Comment by muradbu** [comment]
This would be a very cool addition. I'm doing research on previous efforts to bring tRPC to Directus but there's not much to go off from. Trying to see what I can do!

**Comment by abdonrd** [comment]
This sounds very interesting!

**Comment by DasMergo** [comment]


**Reply by rijkvanzanten** [reply]
It's a ton of work to add it in as an alternative to REST and GraphQL, so we wanna make damn sure it's worth doing 😄 So far, this has gotten quite a couple upvotes, but we're watching the industry for a bit before committing to the long-term support and maintenance in core 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by useEffects** [reply]
Hii any update on this?

**Reply by istvano** [reply]
Hi,

Any update on trpc ?
